tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -185,25 +500,17 @@ def __init__(self, target_executors):
     Raises:
       ValueError: If the value is unrecognized (e.g., a nonexistent intrinsic).
     """
-    py_typecheck.check_type(target_executors, dict)
+    py_typecheck.check_subclass(intrinsic_strategy, IntrinsicStrategy)
+    self.intrinsic_strategy = intrinsic_strategy(self)
+    self.intrinsic_strategy.validate_executor_placements(target_executors)
```

Comment:
`validate_executor_placements` is marked as being a class method but used on an instance here. I would consider completely moving the validation into the strategy, and call this in its ctor

Reply:
I believe this is perfectly acceptable Python. PEP makes no mention forbidding this. Also, classmethod being a descriptor implies this is an expected use. I would've even used staticmethod but it needs to be extensible for subclasses

Reply:
Some background motivation for this is that I wanted the IntrinsicStrategy to be as stateless as possible, since all state in e.g. executors should be managed by the parent executor this strategy is attached to. I even originally had all intrinsic methods being classmethods, however that quickly became far too verbose